test/integration/api_listener_integration_test.cc

```diff
@@ -0,0 +1,121 @@
+#include "test/integration/integration.h"
+#include "test/mocks/http/stream_encoder.h"
+#include "test/test_common/environment.h"
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::ReturnRef;
+
+namespace Envoy {
+namespace {
+
+class ApiListenerIntegrationTest : public BaseIntegrationTest,
+                                   public testing::TestWithParam<Network::Address::IpVersion> {
```

Comment:
Can this potentially be based on the HTTP integration tests? I think you could just use that with a config modifier that replaces the listener contents with an API listener?